elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore.rb

```diff
@@ -43,97 +46,156 @@ module HTTP
+                  raise e
+                end
+              else
+                raise e
+              end
+            rescue ::Elasticsearch::Transport::Transport::HostUnreachableError => e
+              logger.error "[#{e.class}] #{e.message} #{e.url}" if logger
 
-                host.delete(:user)     # auth is not supported here.
-                host.delete(:password) # use the headers
+              if @options[:retry_on_failure]
+                logger.warn "[#{e.class}] Attempt #{tries} connecting to #{connection.host.inspect}" if logger
+                if tries <= max_retries
+                  if @options[:reload_on_failure] && pool.alive_urls_count == 0
+                    logger.warn "[#{e.class}] Reloading connections (attempt #{tries} of #{connections.all.size})" if logger
+                    reload_connections!
+                  end
 
-                Connections::Connection.new \
-                  :host => host,
-                  :connection => @manticore
-              },
-              :selector_class => options[:selector_class],
-              :selector => options[:selector]
+                  retry
+                else
+                  logger.fatal "[#{e.class}] Cannot connect to #{connection.host.inspect} after #{tries} tries" if logger
+                  raise e
+                end
+              end
+            rescue Exception => e
+              logger.fatal "[#{e.class}] #{e.message} ()" if logger
+              raise e
+            end
           end
 
-          # Closes all connections by marking them as dead
-          # and closing the underlying HttpClient instances
-          #
-          # @return [Connections::Collection]
-          #
           def __close_connections
-            @connections.each {|c| c.dead! }
-            @connections.all.each {|c| c.connection.close }
+            if @sniffer
+              logger.info("Closing sniffer...") if logger
+              @sniffer.close
+            end
+            logger.info("Sniffer closed.") if logger
+            logger.info("Closing pool") if logger
+            @pool.close # closes adapter as well
           end
 
-          # Returns an array of implementation specific connection errors.
-          #
-          # @return [Array]
-          #
-          def host_unreachable_exceptions
-            [
-              ::Manticore::Timeout,
-              ::Manticore::SocketException,
-              ::Manticore::ClientProtocolException,
-              ::Manticore::ResolutionFailure
-            ]
+          def enriching_response(method, path, params, body)
+            start = Time.now if logger || tracer
+
+            url, response = yield
+
+            duration = Time.now-start if logger || tracer
+
+            if response.status.to_i >= 300
+              __log    method, path, params, body, url, response, nil, 'N/A', duration if logger
```

Comment:
I think there might be an excess of error messages if a request fails permanently: it will log 1 message for each failed try, and then log here again, right?